arlo-client/src/components/AuditForms/StatusBox.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Callout, H4 } from '@blueprintjs/core'
+import { IAudit } from '../../types'
+import FormButton from '../Form/FormButton'
+
+const Wrapper = styled(Callout)`
+  .details {
+    width: 50%;
+  }
+  .wrapper {
+    margin: 10px 0 10px 0;
+    text-align: right;
+  }
+`
+
+const generateStatuses = (
+  audit: IAudit,
+  launched: boolean,
+  started: boolean,
+  complete: boolean
+): [string, string | undefined] => {
+  const STATUSES = {
+    audit: [
+      'The audit has not launched.',
+      'The audit has not started.',
+      `Round ${audit.rounds.length} of the audit is in progress.`,
+      'The audit is complete.',
+    ],
+    setup: [
+      'Audit setup is not complete.',
+      'Audit setup is complete.',
+      undefined,
+    ],
+  }
+
+  if (complete) {
+    // all jurisdictions have completed the audit
+    return [STATUSES.audit[3], STATUSES.setup[2]]
```

Comment:
using status indexes here is fragile and hard to follow, what if you add one? I'd recommend using objects, e.g. `STATUSES.audit.in_progress`.

Reply:
Definitely agree with @benadida here.

Reply:
Or just remove the indirection altogether:

```ts
const generateStatuses = (
  audit: IAudit,
  launched: boolean,
  started: boolean,
  complete: boolean
): [string, string | undefined] => {
  if (complete) {
    // all jurisdictions have completed the audit
    return ['The audit is complete.', undefined]
  }
  if (launched && started) {
    // rounds have started
    return [
      `Round ${audit.rounds.length} of the audit is in progress.`,
      undefined,
    ]
  }
  if (launched && !started) {
    // after setup before first round
    return ['The audit has not started.', 'Audit setup is complete.']
  }
  /* istanbul ignore next */
  return ['The audit has not launched.', 'Audit setup is not complete.']
}
```

Reply:
@eventualbuddha and @benadida I agree. I initially set it up as the array structure to help me visualize the flow, but you are right that it's better to do it without in the final product.